StylableSwiftUI/Classes/StylistIdentifier.swift

```diff
@@ -137,46 +73,64 @@ extension StylistIdentifier: ExpressibleByStringLiteral {
     }
 }
 
-extension StylistIdentifier: Comparable {
+extension StylistIdentifier {
 
-    /// Returns true if `identifier` is a more specific version of `self`
-    ///
-    /// `*/specific/identifier` matches `very/specific/identifier`
-    func matches(_ identifier: StylistIdentifier) -> Bool {
-        guard self != identifier else { return true }
+    public static var unique: StylistIdentifier { StylistIdentifier(UUID().uuidString) }
+}
+
+
+// MARK: - Component
+
+extension StylistIdentifier {
 
-        // */*/* is a special case (technically it has no identifier) and it matches _everything_
-        guard !self.isWildcard else { return true }
+    public struct Path: CustomStringConvertible, LosslessStringConvertible, ExpressibleByStringLiteral, Equatable, Hashable {
 
-        // Loop through the components, comparing them
-        let count = max(self.components.count, identifier.components.count)
-        for index in 0..<count {
-            let lhs = self.component(at: index)
-            let rhs = identifier.component(at: index)
+        let components: [Component]
 
-            if !lhs.matches(rhs) { return false }
+        init(components: [Component]) {
+            self.components = components
         }
 
-        return true
-    }
+        public init(_ value: String) {
+            self.components = value
+                .split(separator: "/")
+                .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+                .filter { !$0.isEmpty }
+                .reversed()
+                .map { Component($0) }
+        }
 
-    /// `<` here means `lhs` is definitely more specific than `rhs` - `lhs` matches _less_ than `rhs`
-    static public func < (lhs: StylistIdentifier, rhs: StylistIdentifier) -> Bool {
-        return lhs.specificity > rhs.specificity
-    }
-}
+        public init(stringLiteral value: String) {
+            self.init(value)
+        }
 
-// MARK: - Component
+        public var description: String {
+            self.components.reversed().map(String.init).joined(separator: "/")
+        }
 
-extension StylistIdentifier {
+        var isEmpty: Bool { self.components.isEmpty }
+
+        func component(at index: Int) -> Component? {
+            guard index < self.components.count else { return nil }
+            return self.components[index]
+        }
+
+        func within(_ path: Path?) -> Path {
+            guard let path = path else { return self }
```

Comment:
what's the point of having an optional Path here if we're just gong to return self? Doesn't it make more sense to not allow nil values?

Reply:
And if we're optmizing, should we check the given path is not empty?

Reply:
(I think) It makes some of the callers lives easier - I'll check. And if we keep it then we should definitely check to see if it's empty.

Reply:
Only used in one place - so I removed the optionality.

Reply:
^ Found more places with use it, so put the optional back in and added your isEmpty optimisation check :)